documentation/src/main/asciidoc/topics/cache_api.adoc

```diff
@@ -192,26 +195,25 @@ These are, in order of cost:
 * writing to a cache store (optional)
 * locking
 
-As of {brandname} 4.0, using the async methods will take the network calls and marshalling off the critical path.  For various technical reasons, writing to a cache store and acquiring locks, however, still happens in the caller's thread.  In future, we plan to take these offline as well.  See link:http://lists.jboss.org/pipermail/infinispan-dev/2010-January/002219.html[this developer mail list thread] about this topic.
+Using the async methods will take the network calls and marshalling off the critical path.  For various technical reasons, writing to a cache store and acquiring locks, however, still happens in the caller's thread.
 
+//dnaro: ISPN-11577
+ifndef::productized[]
 === Notifying futures
```

Comment:
this section can be removed IMO. We return CompletableFuture (a JDK class) and the users can use any of the methods available to register listeners.

Reply:
removed. thanks @pruivo